.circleci/config.yml

```diff
@@ -116,6 +116,24 @@ jobs:
           path: /tmp/circular_deps.svg
           destination: /hpx/artifacts/circular_deps.svg
 
+  # Insure that the modules stay clang-formatted
+  clang_format:
+    <<: *defaults
+    steps:
+      - attach_workspace:
+          at: /hpx
+      - run:
+          name: Check that the modules are well clang-formatted
+          command: |
+              cd /hpx/source/libs && shopt -s globstar # to activate the ** globbing
+              clang-format-8 -i **/*.{cpp,hpp}
+              git diff --exit-code --name-status > /tmp/modified_format_files.txt
+              if [[ $(git diff --exit-code --name-status) ]]; then exit 1; fi
```

Comment:
Do we really have to execute git twice?

Reply:
@aurianer I *think* you can leave out both the `if` and the `cd /` below. The working directory should get reset for each step. And if `git diff ...` is the last line in the step the exit code of that will be used for the whole step (the redirection doesn't affect it).

Reply:
Yeah I thought that the exit code would be the one of the redirection but it's not !